src/types/include/module_hm_table.hpp

```diff
@@ -2,40 +2,46 @@
 #define MODULE_HM_TABLE
 
 #include "module_error_action.hpp"
-
-
-class ModuleHMTable
-{
-    private:
-      ModuleErrorAction moduleErrorAction[100];
-      monotonic_buffer_resource moduleErrorActionSrc{std::data(moduleErrorAction),
-                                                    std::size(moduleErrorAction)};
-      identifier_t stateIdentifier;       /* required */
-      description_t description;          /* required */
-      vector<ModuleErrorAction> actions;  /* required */
-
-    public:
-      ModuleHMTable() {};
-
-      ModuleHMTable(identifier_t state, std::string descr, std::initializer_list<ModuleErrorAction> actions):
-        stateIdentifier(state), actions(actions, &moduleErrorActionSrc)
-        {
-          strcpy(description, descr.c_str());
+#include <string.h>
+
+class ModuleHMTable {
+private:
+    ModuleErrorAction moduleErrorAction[100];
+    std::vector<ModuleErrorAction>* moduleErrorActions =
+        new (&moduleErrorAction) std::vector<ModuleErrorAction>;
+
+    identifier_t stateIdentifier;      /* required */
+    description_t description;         /* required */
+    vector<ModuleErrorAction> actions; /* required */
+
+public:
+    ModuleHMTable(){};
+
+    ModuleHMTable(identifier_t state, std::string descr, std::initializer_list<ModuleErrorAction> actions)
+        : stateIdentifier(state)
+    {
+        for (auto a : actions) {
+            moduleErrorActions->push_back(a);
         }
-
-      ModuleHMTable(const ModuleHMTable& rhs):
-        stateIdentifier(rhs.stateIdentifier), actions(rhs.actions)
-        {
-          strcpy(description, rhs.description);
+        strcpy(description, descr.c_str());
+    }
+
+    ModuleHMTable(const ModuleHMTable& rhs)
+        : stateIdentifier(rhs.stateIdentifier), actions(rhs.actions)
```

Comment:
Delete: , actions(rhs.actions)
```suggestion
        : stateIdentifier(rhs.stateIdentifier)
```